tests/accessor/accessor_default_values.h

```diff
@@ -0,0 +1,149 @@
+            sycl::host_accessor<T, expected_dims,
+                                std::is_const_v<T>
+                                    ? sycl::access_mode::read
+                                    : sycl::access_mode::read_write>>);
+  }
+}
+
+/**
+ * @brief Provides functor that lets verify that generic accessor's template
+ *        parameters have expected default values
+ * @tparam T Current data type
+ * @param type_name Current data type string representation
+ */
+template <typename T>
+void test_for_generic_acc(const std::string& type_name) {
+  auto section_name =
+      "Verify default value for dimensions template parameter for " +
+      type_name + " data type. [generic accessor]";
+  SECTION(section_name) {
+    REQUIRE(
+        std::is_same_v<sycl::accessor<T>, sycl::accessor<T, expected_dims>>);
+  }
+  section_name = "Verify default value for accessMode template parameter for " +
+                 type_name + " data type. [generic accessor]";
+  SECTION(section_name) {
+    REQUIRE(
+        std::is_same_v<sycl::accessor<T, expected_dims>,
+                       sycl::accessor<T, expected_dims,
+                                      std::is_const_v<T>
+                                          ? sycl::access_mode::read
+                                          : sycl::access_mode::read_write>>);
+  }
+  section_name = "Verify default value for accessMode template parameter for " +
+                 type_name + " data type. [generic accessor]";
+  SECTION(section_name) {
+    REQUIRE(
+        std::is_same_v<sycl::accessor<T, expected_dims>,
+                       sycl::accessor<T, expected_dims,
+                                      std::is_const_v<T>
+                                          ? sycl::access_mode::read
+                                          : sycl::access_mode::read_write>>);
+  }
+  section_name =
+      "Verify default value for accessTarget template parameter for " +
+      type_name + " data type. [generic accessor]";
+  SECTION(section_name) {
+    REQUIRE(std::is_same_v<
+            sycl::accessor<T, expected_dims,
+                           std::is_const_v<T> ? sycl::access_mode::read
+                                              : sycl::access_mode::read_write>,
+            sycl::accessor<T, expected_dims,
+                           std::is_const_v<T> ? sycl::access_mode::read
+                                              : sycl::access_mode::read_write,
+                           expected_target>>);
+  }
+  section_name =
+      "Verify default value for isPlaceholder template parameter for " +
+      type_name + " data type. [generic accessor]";
+  SECTION(section_name) {
+    REQUIRE(std::is_same_v<
+            sycl::accessor<T, expected_dims,
+                           std::is_const_v<T> ? sycl::access_mode::read
+                                              : sycl::access_mode::read_write,
+                           expected_target>,
```

Comment:
It will not cover the case of implementation having different default values for different specializations.
Are you sure we shouldn't cover all possible combinations of template parameters (dimensions and access modes)?

Reply:
coverage for different `access_mode`, `dimension` and `target has been implemented [here](https://github.com/KhronosGroup/SYCL-CTS/pull/335/commits/c040cc7560db878fb467d794a504c6e5e20a0d0e#diff-9ed3eae8f2889bc04dfc0fad247a8fa4e381e1fff4144e0f1a23670d108f6fe9R146-R155)